Source/Scene/processPbrMetallicRoughness.js

```diff
@@ -531,29 +549,67 @@ define([
         fragmentShader += '    vec3 baseColor = baseColorWithAlpha.rgb;\n';
 
         if (hasNormals) {
-            // Add metallic-roughness to fragment shader
-            if (defined(generatedMaterialValues.u_metallicRoughnessTexture)) {
-                fragmentShader += '    vec3 metallicRoughness = texture2D(u_metallicRoughnessTexture, ' + v_texcoord + ').rgb;\n';
-                fragmentShader += '    float metalness = clamp(metallicRoughness.b, 0.0, 1.0);\n';
-                fragmentShader += '    float roughness = clamp(metallicRoughness.g, 0.04, 1.0);\n';
-                if (defined(generatedMaterialValues.u_metallicFactor)) {
-                    fragmentShader += '    metalness *= u_metallicFactor;\n';
-                }
-                if (defined(generatedMaterialValues.u_roughnessFactor)) {
-                    fragmentShader += '    roughness *= u_roughnessFactor;\n';
+            if (useSpecGloss) {
+                if (defined(generatedMaterialValues.u_specularGlossinessTexture)) {
+                    fragmentShader += '    vec4 specularGlossiness = SRGBtoLINEAR4(texture2D(u_specularGlossinessTexture, ' + v_texcoord + '));\n';
+                    fragmentShader += '    vec3 specular = specularGlossiness.rgb;\n';
+                    fragmentShader += '    float glossiness = specularGlossiness.a;\n';
+                    if (defined(generatedMaterialValues.u_specularFactor)) {
+                        fragmentShader += '    specular *= u_specularFactor;\n';
+                    }
+                    if (defined(generatedMaterialValues.u_glossinessFactor)) {
+                        fragmentShader += '    glossiness *= u_glossinessFactor;\n';
+                    }
+                } else {
+                    if (defined(generatedMaterialValues.u_specularFactor)) {
+                        fragmentShader += '    vec3 specular = clamp(u_specularFactor, vec3(0.0), vec3(1.0));\n';
+                    } else {
+                        fragmentShader += '    vec3 specular = vec3(1.0);\n';
+                    }
+                    if (defined(generatedMaterialValues.u_glossinessFactor)) {
+                        fragmentShader += '    float glossiness *= clamp(u_glossinessFactor, 0.0, 1.0);\n';
```

Comment:
You have `float glossiness *=` here, that doesn't compile.

Reply:
I tried fixing this locally by removing the `*`.  This works and shows my SpecGlossVsMetalRough new test model correctly!